internal/cmd/controller/reconciler/bundle_controller.go

```diff
@@ -566,6 +571,52 @@ func (r *BundleReconciler) getOCIReference(ctx context.Context, bundle *fleet.Bu
 	return fmt.Sprintf("oci://%s/%s:latest", string(ref), bundle.Spec.ContentsID), nil
 }
 
+// cloneConfigMap clones a config map, identified by the provided name and
+// namespace, to the namespace of the provided bundle deployment bd. This makes
+// the config map available to agents when deploying bd to downstream clusters.
+func (r *BundleReconciler) cloneConfigMap(
+	ctx context.Context,
+	namespace string,
+	name string,
+	bd *fleet.BundleDeployment,
+) error {
+	namespacedName := types.NamespacedName{
+		Namespace: namespace,
+		Name:      name,
+	}
+	var cm corev1.ConfigMap
+	if err := r.Get(ctx, namespacedName, &cm); err != nil {
+		return fmt.Errorf("failed to load source config map, cannot clone into %q: %w", namespace, err)
+	}
+	// clone the config map, and just change the namespace so it's in the target's namespace
+	targetCM := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cm.Name,
+			Namespace: bd.Namespace,
+		},
+		Data:       cm.Data,
+		BinaryData: cm.BinaryData,
+	}
+
+	if err := controllerutil.SetControllerReference(bd, targetCM, r.Scheme); err != nil {
+		return err
+	}
+
+	updated := targetCM.DeepCopy()
+	if _, err := controllerutil.CreateOrUpdate(ctx, r.Client, targetCM, func() error {
+		targetCM.Data = updated.Data
+		targetCM.BinaryData = updated.BinaryData
+
+		return nil
+	}); err != nil {
+		if !apierrors.IsAlreadyExists(err) {
```

Comment:
Is this a leftover from the previous implementation?
I think `CreateOrUpdate` can't return an _AlreadyExists_ error